### ISSUE TITLE
remove-berry branch issues

### ISSUE THREAD

**Issue by timcortesi** [EscherLabs/Graphene#133]
I am using this single issue to track any issues found in the remove-berry branch.

- [x] In the API Gateway, when saving an API, it used to say "Saving", followed by "Successfully Saved".  Now, it says saving, but there is no additional followup message. Example: http://portalawsdev.binghamton.edu/admin/apiserver/hermesdp/apis/46#routes
  - this seems to be a change in toastr - it won't display that message there even though its being called - I need to understand why
- [x] When editing the routes in the API of the APIGateway, an error is thrown during regex validation.  Example: http://portalawsdev.binghamton.edu/admin/apiserver/hermesdp/apis/46#routes
```javascript
      {label: 'Function Name', name:'function_name', required:true, validate:[{type:'custom',test:e =>{
        if(!/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/.test(evalue)) { // This should be e.value, not evalue
          return  'API name must be a valid php function name';
        }
```
- [x] When editing App Instance Resources, you should not be able to click the minus (-) button to remove the resource from the list.  (I don't think?)  All resources should be required.  Example: http://portalawsdev.binghamton.edu/admin/appinstances/12
- [x] There's a debugger; in the blade template here: (you'll need to view the page source to see it) http://portalawsdev.binghamton.edu/admin/apiserver/hermesdp/environments
- [x] In the API Gateway, you can't edit an existing environment, as it throws a server side error.  (I confirmed that this works correctly in the current production deployment)  Note that I cannot modify the "timtest" environment here: http://portalawsdev.binghamton.edu/admin/apiserver/hermesdp/environments
- [x] MicroApp Instance Resources don't pre-populate correctly.  Example: http://portalawsdev.binghamton.edu/admin/appinstances/12 click on the Resources tab and note that it says "2" instead of the actual resource.
- [x] The "cancel" button doesn't work when publishing a MicroApplication or Workflow.  Example: http://portalawsdev.binghamton.edu/admin/apps/6#, http://portalawsdev.binghamton.edu/admin/workflows/64#
- [x] The "cancel" button doesn't work when importing a MicroApplication or Workflow.  Example: http://portalawsdev.binghamton.edu/admin/apps/6#, http://portalawsdev.binghamton.edu/admin/workflows/64#
- [x] Can't save workflow instance or micro app instance with version of "Latest Working of Published" (-1) with the following error: message: "SQLSTATE[22003]: Numeric value out of range: 1264 Out of range value for column 'workflow_version_id' at row 1 Example: http://portalawsdev.binghamton.edu/admin/workflowinstances/75, http://portalawsdev.binghamton.edu/admin/appinstances/12.  (This error is not when you change the version, but when you go to save the instance with a version already set to -1) 
- [x] There are a couple debugger; lines in grapheneAppEngine: https://github.com/EscherLabs/Graphene/blob/remove-berry-dependencies/public/assets/js/appEngines/grapheneAppEngine.js (I also see that this got moved into an appEngine folder with some alternate app engines... is that live in this branch?
- [x] Some debugger; lines here too: https://github.com/EscherLabs/Graphene/blob/remove-berry-dependencies/public/assets/js/cob/uapp.cob.js
- [x] Changing Workflow Versions does not work correctly.  ('undefined' is sent as the workflow_id in the ajax versions request, and a 404 error is returned) Example: http://portalawsdev.binghamton.edu/admin/groups/60/workflowinstances
- [x] The "second" user you select in the admin users view does not work.  Steps: Search for a user.  Select a user.  Search for a different user.  Select that user.  Note that the user doesn't show up until you refresh the page.  Example: http://portalawsdev.binghamton.edu/admin/users
- [x] Editing an image throws a server side error, as the group_id sent as part of the ajax request is an empty string "".  Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] The "List In Menu" text when editing links includes a bunch of extra HTML in the datagrid.  Example: http://portalawsdev.binghamton.edu/admin/groups/3/links.
- [x] An error is thrown after uploading an image to the dropzone: "Uncaught TypeError: modal is not a constructor" Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] When you click on an existing image in the data grid (to view the image), the same error as the example above is thrown.  Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] Group sort does not work correctly.  Example: http://portalawsdev.binghamton.edu/admin/groups (The  ajax URL appears to be malformed, and a 404 response is returned)
- [x] App Instance sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/appinstances
- [x] Page Sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/pages
- [x] Workflow Sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/workflowinstances
- [x] There is an extra "debugger;" on line 28 of users.js Example: http://portalawsdev.binghamton.edu/admin/groups/3/links
- [x] There is a reference to Berry on lines 29 and 30 of users.js Example: http://portalawsdev.binghamton.edu/admin/groups/3/links
- [x] There is a "debugger;" on line 110 of site.js.  Example: http://portalawsdev.binghamton.edu/admin/sites/1
- [x] Editing user admin types does not appear to work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/316/admins (The user_id does not appear to be passed in the ajax request and a 302 redirect is incorrectly triggered)
- [x] Unable to create/edit/delete API Accounts.  (Pretty sure it's using the wrong route?) Example: http://portalawsdev.binghamton.edu/admin/api_users 

**Other comments**
In general, the upload/download buttons should probably be removed from all administrative data grids.  I'm not sure that administrative functions are advisable in bulk considering the undoable damage which can occur if data is malformed.  (upload/download buttons DO exist in the current berryTables version of Graphene, so maybe the removal of this function should wait for a future update)

**Comment by timcortesi** [comment]
Testing Status

- [x] There are a couple debugger; lines in grapheneAppEngine: https://github.com/EscherLabs/Graphene/blob/remove-berry-dependencies/public/assets/js/appEngines/grapheneAppEngine.js (I also see that this got moved into an appEngine folder with some alternate app engines... is that live in this branch?
- [x] Some debugger; lines here too: https://github.com/EscherLabs/Graphene/blob/remove-berry-dependencies/public/assets/js/cob/uapp.cob.js
- [x] Changing Workflow Versions does not work correctly.  ('undefined' is sent as the workflow_id in the ajax versions request, and a 404 error is returned) Example: http://portalawsdev.binghamton.edu/admin/groups/60/workflowinstances
- [x] The "second" user you select in the admin users view does not work.  Steps: Search for a user.  Select a user.  Search for a different user.  Select that user.  Note that the user doesn't show up until you refresh the page.  Example: http://portalawsdev.binghamton.edu/admin/users
- [x] Editing an image throws a server side error, as the group_id sent as part of the ajax request is an empty string "".  Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] The "List In Menu" text when editing links includes a bunch of extra HTML in the datagrid.  Example: http://portalawsdev.binghamton.edu/admin/groups/3/links.
- [x] An error is thrown after uploading an image to the dropzone: "Uncaught TypeError: modal is not a constructor" Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] When you click on an existing image in the data grid (to view the image), the same error as the example above is thrown.  Example: http://portalawsdev.binghamton.edu/admin/groups/60/images
- [x] Group sort does not work correctly.  Example: http://portalawsdev.binghamton.edu/admin/groups (The  ajax URL appears to be malformed, and a 404 response is returned)
- [x] App Instance sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/appinstances
- [x] Page Sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/pages
- [x] Workflow Sort does not work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/60/workflowinstances
- [x] There is an extra "debugger;" on line 28 of users.js Example: http://portalawsdev.binghamton.edu/admin/groups/3/links
- [x] There is a reference to Berry on lines 29 and 30 of users.js Example: http://portalawsdev.binghamton.edu/admin/groups/3/links
- [x] There is a "debugger;" on line 110 of site.js.  Example: http://portalawsdev.binghamton.edu/admin/sites/1
- [x] Editing user admin types does not appear to work correctly. Example: http://portalawsdev.binghamton.edu/admin/groups/316/admins (The user_id does not appear to be passed in the ajax request and a 302 redirect is incorrectly triggered)
- [x] Unable to create/edit/delete API Accounts.  (Pretty sure it's using the wrong route?) Example: http://portalawsdev.binghamton.edu/admin/api_users

**Comment by timcortesi** [comment]
@Cloverstone 

All issues raised have been verified fixed (see above).  (Note: There continued to be issues with redirecting trailing slashes on some APIs, but I addressed that separately by removing the apache redirect in the default .htaccess file.  I also removed a bunch more random debugger; lines)

There is one remaining issue I can see specific to displaying Image URLs:

**Note the difference between the images below:**

<img width="604" alt="Screen Shot 2021-06-18 at 4 04 16 PM" src="https://user-images.githubusercontent.com/5395437/122611315-19df7200-d04f-11eb-9174-d8a8ccf2940a.png">
<img width="604" alt="Screen Shot 2021-06-18 at 4 04 43 PM" src="https://user-images.githubusercontent.com/5395437/122611318-1a780880-d04f-11eb-8128-34b78f9779a5.png">

The URL code appears to exist, but it isn't being displayed.
